test-common/src/main/java/de/bwaldvogel/mongo/backend/AbstractOplogTest.java

```diff
@@ -456,4 +463,66 @@ public void testOplogShouldFilterNamespaceOnChangeStreams() throws Exception {
         return subscriber.values().get(0);
     }
 
+    @Test
+    public void testMultipleChangeStreams() throws InterruptedException {
+        Flowable.fromPublisher(asyncCollection.insertOne(json("_id: 1")))
+            .test().awaitDone(5, TimeUnit.SECONDS).assertComplete();
+
+        final int changeStreamCount = 32;
+
+        List<Bson> pipeline = singletonList(match(Filters.eq("fullDocument.bu", "abc")));
+
+        final TestSubscriber<Map<Integer, List<ChangeStreamDocument<Document>>>> streamSubscriber
+            = new TestSubscriber<>();
+
+        Flowable.range(1, changeStreamCount)
+            .flatMapSingle(index -> {
+                return Flowable.fromPublisher(asyncCollection.watch(pipeline))
+                    .take(2)
+                    .toList()
+                    .map(changeStreamDocuments -> {
+                        return new AbstractMap.SimpleEntry<>(index, changeStreamDocuments);
+                    })
+                    .subscribeOn(Schedulers.io()); // subscribe to change streams concurrently
+            })
+            .toMap(Map.Entry::getKey, Map.Entry::getValue)
+            .toFlowable()
+            .subscribe(streamSubscriber);
+
+        // give time for all ChangeStream Publishers to be subscribed to
+        // todo: expose API to get cursors from Backend and wait until 'changeStreamCount' cursors
+        TimeUnit.SECONDS.sleep(5);
```

Comment:
A required sleep in a test screams for "it will break eventually". I’m sorry, but IMO it’s not acceptable to merge such a test.
Typically one uses something like a `CyclicBarrier` to make such concurrency tests fully deterministic and get rid of sleeps. Please also see my other comments.